### ISSUE TITLE
browser tab crashes with "out of memory"

### ISSUE THREAD

**Issue by ryandague** [Genuitec/CodeTogether#136]
**Describe the bug**
Browser tab crashes within few seconds after connecting. This happens every time.

**To Reproduce**
1. Start a session.
2. Join as a participant.
3. Session appears to function correctly for a bit, usually around 5 to 30 seconds.
4. Participants browser session crashes to the "Aw, Snap!" screen in Chrome with Error code: Out of Memory.

**Expected behavior**
Not crashing

**CodeTogether Host**
 - IDE: Eclipse
 - IDE Version Version: 2020-06 (4.16.0)
 - OS: Windows 10
 - OS Version: 1909 18363.900

**Browser Participant**
 - Browser: Chrome 
 - Browser Version: Version 83.0.4103.116 (Official Build) (64-bit)
 - OS: Windows 10
 
**Additional context**
I created a new Eclipse workspace with a small test project and it works correctly. This only seems to happen on our main project, which is quite large.

**Comment by TimWebb** [comment]
Odd! Would you be able to give any other data about the type of project?

Are there thousands of files in the root of the project, or in the folders you have open?

Are the source files particularly large?

**Comment by ryandague** [comment]
Yes to both. There are about 6500 files in the root folder of the project. Also, we have some source files with 100k+ lines in them. It's definitely not an ideal project structure, but unfortunately I am unable to change that currently.
